### ISSUE TITLE
GetAll<T> InScope

### ISSUE THREAD

**Issue by mcopjan** [ninject/Ninject#356]
This is more a question as I have been stuck on this for a while.

I have the following binding 
``
Bind<IClient>().To<Safari>().When(b => configuration.Target.Equals(Target.Safari)).InScope(ctx => NUnit.Framework.Internal.TestExecutionContext.CurrentContext.CurrentTest.Id);
``

Sometimes when creating of a new instance of Safari fails I am still stepping into test TearDown and making sure I close client instance using

``
kernel.Get<IClient>().Quit();
``
and
``
kernel.Get<IClient>().Dispose();
``

if there is no instance of IClient in scope, Ninject correctly tries to activate one . **I want to prevent that.** So I was hoping to get my instance from the list of activated instances using
``
kernel.GetAll<IClient>() 
``
but I need to somehow filter this list and **get activated instance InScope()** as there may be multiple instances in the list when tests are running in parallel. Is there a way to do so? Or any other way really? Any help would be much appreciated.

**Comment by drieseng** [comment]
Do you want to just dispose the **IClient** instance, or any service created in the scope?
You probably know this but once the scope is finalized, any instances in that scope will be released as well.
If you enable the disposable strategy, any instances that implement **IDisposable** will be disposed when the scope is finalized.
However, I suppose you don't want to rely on the GC to kick in.

**Comment by mcopjan** [comment]
@drieseng yes I want to dispose IClient instance, the problem is **I still don't know how to get IClient instance in my test scope**.
Let me give you a bit more details about my scenario:
Let' say I run 3 tests in parallel, each of them will activate client on first kernel.Get<IClient>() call, any other call will give me **instance of client per test**. Now ideally if all 3 tests activated client successfully, then kernel.GetAll<IClient>() will give me a list of 3 client instances. Let's assume one of the test failed on activation (environment is down, etc.), so in teardown 2 tests will call kernel.Get<IClient>() which in the test scope returns existing instances, i will then call Close() and Dispose() (these are Selenium methods, Dispose() will also remove them from Ninject kernel though). 
3rd test though failed on activation, therefore there is no instance in test scope meaning kernel will try to activate again in TearDown which I want to prevent.

**Comment by drieseng** [comment]
As I said, the easiest approach right now is to enable the disposable strategy, and clear the scope in the TearDown method.  Let me know if you want some sample code for this.

**Comment by mcopjan** [comment]
@drieseng a sample code would be appreciated thanks. For now the way I resolved my problem was moving activation of client from its constructor so when I get instance of client I can check whether it was activated or not based on client property.
